---
title: "RNA_TCGA_BEX"
output: html_document
date: "2024-01-23"
editor_options: 
  chunk_output_type: console
---
########

Code made for RNA431 docker
To build TCGA-TALL obj in an RNA docker and not with Seurat

########


```{r setup, include=FALSE}
library(TCGAbiolinks)
library(DESeq2)
install.packages("tidyverse")
library(tidyverse) #load all tidyverse suite : ggplot2,dplyr...
library(DT)
```


Set your working directory to the folder clone from github
# Fetch data
```{r}
setwd("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/")
```

## Target AML

When I fetch the data for the first time I had an issue with duplicated samples, so I just remove them from the fetching code to correct this problem : 
```{r}
Proj = "TARGET-AML"
query_mRNA_AML <- GDCquery(
    project = Proj,
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification"
  )


#Issue with duplicated samples
AML_sample_subset <- sample(query_mRNA_AML[[1]][[1]]$cases)

AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARSAN-03A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PADYIR-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PADYIR-09A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAECCE-09A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAECCE-09A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAEERJ-09A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAEERJ-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAKIWK-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAKIWK-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAKVGI-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAKVGI-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAPXRJ-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PAPXRJ-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARDDY-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARDDY-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARFAL-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARFAL-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARIMT-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARIMT-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARJCR-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARJCR-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARPDS-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARPDS-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARSAN-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARSAN-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARTAL-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARTAL-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARUBT-40A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARUBT-40A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARUNX-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARUNX-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARYFN-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARYFN-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARYFN-04A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARYFN-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARYVW-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PARYVW-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASBHI-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASBHI-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASCRZ-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASCRZ-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASGZS-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASGZS-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASHBI-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASHBI-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASHYZ-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASHYZ-04A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASIBG-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASIBG-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASJGZ-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASJGZ-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASJTM-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASJTM-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASLSD-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASLSD-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASMGW-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASMGW-09A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASTTW-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASTTW-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASTUH-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASTUH-03A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASTUH-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASTUH-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASVVS-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASVVS-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASVYA-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASVYA-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASVYA-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASVYA-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASVYL-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASVYL-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASWAT-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASWAT-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASWLN-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASWLN-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASXNR-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PASXNR-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATDHA-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATDHA-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATIAK-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATIAK-04A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATIAK-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATIAK-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATJHJ-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATJHJ-09A-01R"] 
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATJHJ-40A-01R"]
AML_sample_subset <- AML_sample_subset[!AML_sample_subset== "TARGET-20-PATJHJ-40A-01R"]
```


```{r}
#https://portal.gdc.cancer.gov/repository?facetTab=cases&filters=%7B"op"%3A"and"%2C"content"%3A%5B%7B"op"%3A"in"%2C"content"%3A%7B"field"%3A"cases.project.project_id"%2C"value"%3A%5B"TARGET-AML"%5D%7D%7D%2C%7B"op"%3A"in"%2C"content"%3A%7B"field"%3A"files.access"%2C"value"%3A%5B"open"%5D%7D%7D%2C%7B"op"%3A"in"%2C"content"%3A%7B"field"%3A"files.data_type"%2C"value"%3A%5B"Gene%20Expression%20Quantification"%5D%7D%7D%2C%7B"op"%3A"in"%2C"content"%3A%7B"field"%3A"files.experimental_strategy"%2C"value"%3A%5B"RNA-Seq"%5D%7D%7D%5D%7D&searchTableTab=cases
# 3225 files for 2281 AML

##### Transcriptomic raw count data AML. ######
Proj = "TARGET-AML"
query_mRNA_AML <- GDCquery(
    project = Proj,
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification",
    barcode= AML_sample_subset
  )

GDCdownload(query_mRNA_AML)
expdat_mRNA_AML <- GDCprepare(query = query_mRNA_AML) # Preparation of matrix


###### CLINICAL DATA ######
#FAB Category
# We are not interested in all AML the same way, having all subtype is of interest but not the 2000 we'll enrich in M0 as they are closer to ALAL and ETP
query_clinical_AML <- GDCquery(
    project = Proj,
    data.category = "Clinical"
)
GDCdownload(query_clinical_AML)

#Files are a mess, will combine all of them and just use the line corresponding to the sample I got
Target_clinical_AML1 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-AML/Clinical/Clinical_Supplement/3bf97830-865d-4e66-aaf1-d5bac81f119a/TARGET_AML_ClinicalData_Discovery_20230720.xlsx")
Target_clinical_AML1 <- Target_clinical_AML1[,c("TARGET USI","FAB Category")] #subsetting the df

Target_clinical_AML2 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-AML/Clinical/Clinical_Supplement/5f8b7137-c1e1-4191-aee5-a5ea55d32ca7/TARGET_AML_ClinicalData_Validation_20230720.xlsx")
Target_clinical_AML2 <- Target_clinical_AML2[,c("TARGET USI","FAB Category")] #subsetting the df

Target_clinical_AML3 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-AML/Clinical/Clinical_Supplement/129e5399-6892-4cb7-99a8-1406309a2e4d/TARGET_AML_ClinicalData_LowDepthRNAseq_20230720.xlsx")
Target_clinical_AML3 <- Target_clinical_AML3[,c("TARGET USI","FAB Category")] #subsetting the df

Target_clinical_AML4 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-AML/Clinical/Clinical_Supplement/727dbb38-32fc-4444-9db2-13c8eb427fa2/TARGET_AML_ClinicalData_AAML1031_AAML0631_additionalCasesForSortedCellsAndCBExperiment_20230720.xlsx")
Target_clinical_AML4 <- Target_clinical_AML4[,c("TARGET USI","FAB Category")] #subsetting the df

Target_clinical_AML5 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-AML/Clinical/Clinical_Supplement/68170d63-c297-4d93-a21b-7bb43b451a96/TARGET_AML_ClinicalData_AML1031_20230720.xlsx")
Target_clinical_AML5 <- Target_clinical_AML5[,c("TARGET USI","FAB Category")] #subsetting the df

Target_clinical_AML <- bind_rows(Target_clinical_AML1, Target_clinical_AML2,Target_clinical_AML3,Target_clinical_AML4,Target_clinical_AML5) %>% distinct(`TARGET USI`, .keep_all = T) #To remove TARGET USI duplicates
#Getting 2144 patient so close to what TCGA internet told us.

patientID_keep <-intersect(Target_clinical_AML$`TARGET USI`,expdat_mRNA_AML$patient) # 1930 cases with RNA and clinical info

Target_clinical_AML<- Target_clinical_AML[Target_clinical_AML$`TARGET USI` %in% patientID_keep,]
table(Target_clinical_AML$`FAB Category`) #We don't need all those AML will downsample but need to keep all M0

Target_clinical_AML<- Target_clinical_AML[Target_clinical_AML$`FAB Category` != "Unknown",] # removing unknown..
Target_clinical_AML<- Target_clinical_AML[Target_clinical_AML$`FAB Category` != "Not classified",] # removing not classifed 

samples_per_group <- 30
# This function will take a maximum of x patient and if less will take all
subsample <- Target_clinical_AML %>%
  group_by(`FAB Category`) %>%
  slice(sample(n(), min(samples_per_group, n()))) %>%
  ungroup()

table(subsample$`FAB Category`)


##################

#Prepare for DESEQ2
assayNames(expdat_mRNA_AML)
raw_counts <- assays(expdat_mRNA_AML)[["unstranded"]] # to be sure to select raw count

exp_RNA <- assay(expdat_mRNA_AML) # To get count matrix
#rename with patient ID
colnames(exp_RNA) <- expdat_mRNA_AML$patient


subsample <- subsample %>% distinct(`TARGET USI`, .keep_all=TRUE)
sampl_info_AML <- as.data.frame(subsample)
sampl_info_AML <- na.omit(sampl_info_AML)
rownames(sampl_info_AML) <- sampl_info_AML$`TARGET USI`
colnames(sampl_info_AML) <- c("TARGET_USI","Classification")

#susbet RNA matrix with only patient ID
exp_RNA_AML<- exp_RNA[,rownames(sampl_info)] 

saveRDS(exp_RNA_AML, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_AML.rds")
saveRDS(sampl_info_AML, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinial_matrix_TARGET_AML.rds")
```

## Target P2
```{r}
# TARGET P2
##### Transcriptomic raw count data P2. ######
Proj = "TARGET-ALL-P2"
query_mRNA_P2 <- GDCquery(
    project = Proj,
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification"
  )
GDCdownload(query_mRNA_P2)
expdat_mRNA_P2 <- GDCprepare(query = query_mRNA_P2) # Preparation of matrix

datatable(
    as.data.frame(colData(expdat_mRNA_P2)), 
    options = list(scrollX = TRUE, keys = TRUE, pageLength = 5), 
    rownames = FALSE
)

rowRanges(expdat_mRNA_P2)


###### CLINICAL DATA ######
# Get primary diagnosis ID

query_clinical_P2 <- GDCquery(
    project = Proj,
    data.category = "Clinical"
  )
GDCdownload(query_clinical_P2)

#Files are a mess, will combine all of them and just use the line corresponding to the sample I got
P2_Target_clinical_ALL1 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-ALL-P2/Clinical/Clinical_Supplement/304aefc8-6a70-4896-9f54-2605d7e15630/TARGET_ALL_ClinicalData_Phase_I_20230727.xlsx")
P2_Target_clinical_ALL2 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-ALL-P2/Clinical/Clinical_Supplement/19889559-1020-4fe3-b81e-b2eaa3f5b16c/TARGET_ALL_ClinicalData_Phase_II_Discovery_20230727.xlsx")
P2_Target_clinical_ALL3 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-ALL-P2/Clinical/Clinical_Supplement/64174210-4250-47b6-a99c-28fae719ab8d/TARGET_ALL_ClinicalData_Dicentric_20230727.xlsx")
P2_Target_clinical_ALL4 <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-ALL-P2/Clinical/Clinical_Supplement/c6a63e40-f4e5-4002-9b87-3e6f3b3dfab4/TARGET_ALL_ClinicalData_Phase_II_Validation_20230727.xlsx") # with B-ALL and T-ALL

P2_Target_clinical_ALL <- bind_rows(P2_Target_clinical_ALL1, P2_Target_clinical_ALL2,P2_Target_clinical_ALL3,P2_Target_clinical_ALL4) %>% distinct(`TARGET USI`, .keep_all = T) #To remove TARGET USI duplicates

# online : 
#https://portal.gdc.cancer.gov/repository?facetTab=files&filters=%7B"op"%3A"and"%2C"content"%3A%5B%7B"content"%3A%7B"field"%3A"cases.project.project_id"%2C"value"%3A%5B"TARGET-ALL-P2"%5D%7D%2C"op"%3A"in"%7D%2C%7B"op"%3A"in"%2C"content"%3A%7B"field"%3A"files.data_type"%2C"value"%3A%5B"Gene%20Expression%20Quantification"%5D%7D%7D%2C%7B"content"%3A%7B"field"%3A"files.experimental_strategy"%2C"value"%3A%5B"RNA-Seq"%5D%7D%2C"op"%3A"in"%7D%5D%7D&searchTableTab=cases
#In P2 and experimental strategy is RNA-seq and data type is Gene expression quantification 
# Online TCGA told us we are supposed to have 532 files and 469 cases

patientID_keep <-intersect(P2_Target_clinical_ALL$`TARGET USI`,expdat_mRNA_P2$patient) # 469 cases so it's a perfect match 


# Get subgroup of T-ALL
# Loading the subgroup classification file
P2_Target_clinical_TALL <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-ALL-P2/Clinical/Clinical_Supplement/206f7eec-326e-4fcd-b244-338a522ff384/TARGET_ALL_ClinicalData_Validation_Supplement_20230727.xlsx")
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[,c("TARGET USI","Group")] #subsetting the df

#P2_Target_clinical_TALL <- P2_Target_clinical_TALL[,c("TARGET USI","Group","ETP status")] #subsetting the df


##################

#Create a experimental design for DESEQ2
exp_RNA <- assay(expdat_mRNA_P2) # To get count matrix
#rename with patient ID
colnames(exp_RNA) <- expdat_mRNA_P2$patient

sampl_info <- P2_Target_clinical_ALL[,c("TARGET USI","Cell of Origin")] #subset df

sampl_info <- sampl_info[sampl_info$`TARGET USI` %in% patientID_keep,] #reduce size to keep only ID with RNA infos

# For T-ALL patient removing from sample info P2 patient with unknow T-ALL subgroup patient
unique(P2_Target_clinical_TALL$Group)
Unk <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "Unknown",]$`TARGET USI`
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[ ! P2_Target_clinical_TALL$`TARGET USI` %in% Unk, ]
Unk <- P2_Target_clinical_TALL[which(is.na(P2_Target_clinical_TALL$Group)),]$`TARGET USI` # Get NA patient
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[ ! P2_Target_clinical_TALL$`TARGET USI` %in% Unk, ]

ID_TALL_detailed <-P2_Target_clinical_TALL$`TARGET USI` # list of patient with T-ALL classification
ID_TALL_all <- sampl_info[sampl_info$`Cell of Origin` == "T Cell ALL",]$`TARGET USI` # ALl T-ALL patient
patient_remove <- setdiff(ID_TALL_all,ID_TALL_detailed)

sampl_info <- sampl_info[ ! sampl_info$`TARGET USI` %in% patient_remove, ]
sampl_info_P2 <- as.data.frame(sampl_info)
rownames(sampl_info_P2) <- sampl_info_P2$`TARGET USI`
colnames(sampl_info_P2) <- c("TARGET_USI","Classification")

patientID_keep <-intersect(sampl_info_P2$`TARGET_USI`,expdat_mRNA_P2$patient) # 469 cases so it's a perfect match 

#susbet RNA matrix with only patient ID
exp_RNA_P2 <- exp_RNA[,patientID_keep]
saveRDS(exp_RNA_P2, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_P2.rds")
saveRDS(sampl_info_P2, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinial_matrix_TARGET_P2.rds")
```


## Target P3
```{r}
# online : 
#https://portal.gdc.cancer.gov/repository?facetTab=files&filters=%7B"op"%3A"and"%2C"content"%3A%5B%7B"content"%3A%7B"field"%3A"cases.project.project_id"%2C"value"%3A%5B"TARGET-ALL-P3"%5D%7D%2C"op"%3A"in"%7D%2C%7B"op"%3A"in"%2C"content"%3A%7B"field"%3A"files.data_type"%2C"value"%3A%5B"Gene%20Expression%20Quantification"%5D%7D%7D%2C%7B"content"%3A%7B"field"%3A"files.experimental_strategy"%2C"value"%3A%5B"RNA-Seq"%5D%7D%2C"op"%3A"in"%7D%5D%7D&searchTableTab=cases
#In P3 and experimental strategy is RNA-seq and data type is Gene expression quantification 
# Online TCGA told us we are supposed to have 135 files and 110 cases

##### Transcriptomic raw count data P2. ######
Proj = "TARGET-ALL-P3"
query_mRNA_P3 <- GDCquery(
    project = Proj,
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification"
  )
GDCdownload(query_mRNA_P3)
expdat_mRNA_P3 <- GDCprepare(query = query_mRNA_P3) # Preparation of matrix

###### CLINICAL DATA ######
# Get primary diagnosis ID

query_clinical_P3 <- GDCquery(
    project = Proj,
    data.category = "Clinical"
  )
GDCdownload(query_clinical_P3)

P3_Target_clinical_ALAL <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-ALL-P3/Clinical/Clinical_Supplement/d475576c-eac8-4f24-bfc7-67797b02b109/TARGET_ALL_ClinicalData_Phase_III_20230725.xlsx")

P3_Target_clinical_ALAL$`WHO ALAL Classification`

patientID_keep <-intersect(P3_Target_clinical_ALAL$`TARGET USI`,expdat_mRNA_P3$patient) # 92 cases we loose some cases compare to internet info
P3_Target_clinical_ALAL<- P3_Target_clinical_ALAL[,c("TARGET USI","WHO ALAL Classification")] #subsetting the df
P3_Target_clinical_ALAL<- P3_Target_clinical_ALAL[P3_Target_clinical_ALAL$`TARGET USI` %in% patientID_keep,]
##################

#Prepare for DESEQ2
exp_RNA <- assay(expdat_mRNA_P3) # To get count matrix
#rename with patient ID
colnames(exp_RNA) <- expdat_mRNA_P3$patient


sampl_info_P3 <- as.data.frame(P3_Target_clinical_ALAL)
rownames(sampl_info_P3) <- sampl_info_P3$`TARGET USI`
colnames(sampl_info_P3) <- c("TARGET_USI","Classification")

#susbet RNA matrix with only patient ID
exp_RNA_P3<- exp_RNA[,rownames(sampl_info_P3)]

saveRDS(exp_RNA_P3, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_P3.rds")
saveRDS(sampl_info_P3, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinial_matrix_TARGET_P3.rds")
```

# Object preparation 
Preparing the matrices
```{r}
clinial_matrix_TARGET_AML <- readRDS(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinial_matrix_TARGET_AML.rds")
clinial_matrix_TARGET_P2  <- readRDS(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinial_matrix_TARGET_P2.rds")
clinial_matrix_TARGET_P3  <- readRDS(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinial_matrix_TARGET_P3.rds")

RNA_matrix_TARGET_AML  <- readRDS(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_AML.rds")
RNA_matrix_TARGET_P2  <- readRDS(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_P2.rds")
RNA_matrix_TARGET_P3  <- readRDS(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_P3.rds")

#Binding all dataset
clinical_matrix_TARGET <- rbind(clinial_matrix_TARGET_AML,clinial_matrix_TARGET_P2,clinial_matrix_TARGET_P3)
RNA_matrix_TARGET <- cbind(RNA_matrix_TARGET_AML,RNA_matrix_TARGET_P2,RNA_matrix_TARGET_P3)

# reorgnaisation medatadata
unique(clinical_matrix_TARGET$Classification)
# The classification column is detailed for AML but not for T and B.. I need to sort this into different columns
colnames(clinical_matrix_TARGET) <- c("TARGET_USI","detailed_class")
clinical_matrix_TARGET$simple_class <- clinical_matrix_TARGET$detailed_class
unique(clinical_matrix_TARGET$simple_class)

clinical_matrix_TARGET <- within(clinical_matrix_TARGET,simple_class[detailed_class %in% c("M0 Undifferentiated","M1","M2","M3","M4","M5","M6","M7")] <- "AML")
clinical_matrix_TARGET <- within(clinical_matrix_TARGET,simple_class[detailed_class %in% c("B-Precursor")] <- "B-ALL")
clinical_matrix_TARGET <- within(clinical_matrix_TARGET,simple_class[detailed_class %in% c("T Cell ALL")] <- "T-ALL")
clinical_matrix_TARGET <- within(clinical_matrix_TARGET,simple_class[detailed_class %in% c("T/M","B/M","NOS (T/B/M)","AUL","NOS (T/B)","MLL")] <- "ALAL")

#Some category I don"t need and filter out
sum(clinical_matrix_TARGET$simple_class == "Ph+", na.rm = TRUE)
sum(clinical_matrix_TARGET$simple_class == "B precursor (Non-T, Non-B ALL)", na.rm = TRUE)
clinical_matrix_TARGET <- clinical_matrix_TARGET[clinical_matrix_TARGET$simple_class != "Ph+",]
clinical_matrix_TARGET <- clinical_matrix_TARGET[clinical_matrix_TARGET$simple_class != "B precursor (Non-T, Non-B ALL)",]

# Get subgroup of T-ALL
# Loading the subgroup classification file
P2_Target_clinical_TALL <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-ALL-P2/Clinical/Clinical_Supplement/206f7eec-326e-4fcd-b244-338a522ff384/TARGET_ALL_ClinicalData_Validation_Supplement_20230727.xlsx")
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[,c("TARGET USI","Group","ETP status")] #subsetting the df

# For T-ALL patient removing from sample info P2 patient with unknow T-ALL subgroup patient
unique(P2_Target_clinical_TALL$Group)
Unk <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "Unknown",]$`TARGET USI`
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[ ! P2_Target_clinical_TALL$`TARGET USI` %in% Unk, ]
Unk <- P2_Target_clinical_TALL[which(is.na(P2_Target_clinical_TALL$Group)),]$`TARGET USI` # Get NA patient
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[ ! P2_Target_clinical_TALL$`TARGET USI` %in% Unk, ]

tal1<- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "TAL1",]$`TARGET USI` # all ID for TAL1 patient
tal2<- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "TAL2",]$`TARGET USI` 
tal3<- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "LMO1/2",]$`TARGET USI` 
NKX1<- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "NKX2_1",]$`TARGET USI`
LYL <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "LMO2_LYL1",]$`TARGET USI` 
HOXA <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "HOXA",]$`TARGET USI` 
TLX1 <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "TLX1",]$`TARGET USI` 
TLX3 <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "TLX3",]$`TARGET USI`

clinical_matrix_TARGET[clinical_matrix_TARGET$TARGET_USI %in% tal1,]$detailed_class <- "TAL/LMO"
clinical_matrix_TARGET[clinical_matrix_TARGET$TARGET_USI %in% tal2,]$detailed_class <- "TAL/LMO"
clinical_matrix_TARGET[clinical_matrix_TARGET$TARGET_USI %in% tal3,]$detailed_class <- "TAL/LMO"
clinical_matrix_TARGET[clinical_matrix_TARGET$TARGET_USI %in% NKX1,]$detailed_class <- "TLX"
clinical_matrix_TARGET[clinical_matrix_TARGET$TARGET_USI %in% LYL,]$detailed_class <- "LYL/LMO"
clinical_matrix_TARGET[clinical_matrix_TARGET$TARGET_USI %in% HOXA,]$detailed_class <- "HOXA"
clinical_matrix_TARGET[clinical_matrix_TARGET$TARGET_USI %in% TLX1,]$detailed_class <- "TLX"
clinical_matrix_TARGET[clinical_matrix_TARGET$TARGET_USI %in% TLX3,]$detailed_class <- "TLX"

unique(clinical_matrix_TARGET$detailed_class)

#subset RNA matrix to match clinical ID 
patientID_keep <-intersect(clinical_matrix_TARGET$TARGET_USI,colnames(RNA_matrix_TARGET)) 
RNA_matrix_TARGET<- RNA_matrix_TARGET[,patientID_keep]

saveRDS(clinical_matrix_TARGET, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinical_matrix_TARGET_AcuteLeuk.rds")
saveRDS(RNA_matrix_TARGET, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_AcuteLeuk.rds")
```

## T-ALL obj
```{r}
RNA_matrix_TARGET <- readRDS("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_AcuteLeuk.rds")
clinical_matrix_TARGET <- readRDS("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinical_matrix_TARGET_AcuteLeuk.rds")

all(rownames(clinical_matrix_TARGET) == colnames(RNA_matrix_TARGET)) #they are all in the same order of patient


## Subset on T-ALL for DGE
ID_TALL <-clinical_matrix_TARGET[clinical_matrix_TARGET$simple_class == "T-ALL",]$`TARGET_USI`

clinical_matrix_TARGET_TALL <- clinical_matrix_TARGET[clinical_matrix_TARGET$`TARGET_USI` %in% ID_TALL, ]
RNA_matrix_TARGET_TALL <- RNA_matrix_TARGET[,ID_TALL]

saveRDS(clinical_matrix_TARGET_TALL, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinical_matrix_TARGET_TALL.rds")

```

### Convert ID
```{r}
# https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/ to convert geneID using TCGA gtf
#GENCODE v36 gtf https://gdc.cancer.gov/about-data/gdc-data-processing/gdc-reference-files

BiocManager::install("rtracklayer")
library(rtracklayer)
gtf.gr = rtracklayer::import('/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/gencode.v36.annotation.gtf') 
gtf.df = as.data.frame(gtf.gr)

gtf.df <- gtf.df[gtf.df$type == "gene",]
gtf.df <- gtf.df[gtf.df$gene_type == "protein_coding",] #keeping only protein coding Gene

#Looking at duplicates
n_occur <- data.frame(table(gtf.df$gene_name))
temp <- n_occur[n_occur$Freq > 1,]

#Modification to add their Ensmbl ID
tmp <- gtf.df[gtf.df$gene_name %in% temp$Var1,]
tmp <- tmp[,c("gene_id","gene_name")] #subsetting the df
tmp$gene_name <- paste(tmp$gene_id,tmp$gene_name,sep="_")

gtf.df[gtf.df$gene_id %in% tmp$gene_id,]$gene_name <- tmp$gene_name

genes = (gtf.df[ ,c("gene_id","gene_name")])

length(unique(genes$gene_id))# so all ensembl ID are unique at this step

genes <- genes[order(genes$gene_id),]


## Subset RNA matrix to keep only protein coding genes
RNA_matrix_TARGET_TALL <- RNA_matrix_TARGET_TALL[rownames(RNA_matrix_TARGET_TALL) %in% genes$gene_id, ]

tmp<- intersect(genes$gene_id,rownames(RNA_matrix_TARGET_TALL))
tmp <- genes[genes$gene_id %in% tmp,]#subset gene to match the genes in my df if needed

all(tmp$gene_id == rownames(RNA_matrix_TARGET_TALL)) #they are all in the same order of genes


rownames(RNA_matrix_TARGET_TALL) <- tmp$gene_name #rename 

saveRDS(RNA_matrix_TARGET_TALL, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_TALL_convert.rds")
```


## Acute Leuk obj
```{r}
RNA_matrix_TARGET <- readRDS("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_AcuteLeuk.rds")
clinical_matrix_TARGET <- readRDS("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinical_matrix_TARGET_AcuteLeuk.rds")

all(rownames(clinical_matrix_TARGET) == colnames(RNA_matrix_TARGET)) #they are all in the same order of patient
```

### Convert ID
```{r}
# https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/ to convert geneID using TCGA gtf
#GENCODE v36 gtf https://gdc.cancer.gov/about-data/gdc-data-processing/gdc-reference-files


BiocManager::install("rtracklayer")
library(rtracklayer)
gtf.gr = rtracklayer::import('/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/gencode.v36.annotation.gtf') 
gtf.df = as.data.frame(gtf.gr)

gtf.df <- gtf.df[gtf.df$type == "gene",]
gtf.df <- gtf.df[gtf.df$gene_type == "protein_coding",] #keeping only protein coding Gene

#Looking at duplicates
n_occur <- data.frame(table(gtf.df$gene_name))
temp <- n_occur[n_occur$Freq > 1,]

#Modification to add their Ensmbl ID
tmp <- gtf.df[gtf.df$gene_name %in% temp$Var1,]
tmp <- tmp[,c("gene_id","gene_name")] #subsetting the df
tmp$gene_name <- paste(tmp$gene_id,tmp$gene_name,sep="_")

gtf.df[gtf.df$gene_id %in% tmp$gene_id,]$gene_name <- tmp$gene_name

genes = (gtf.df[ ,c("gene_id","gene_name")])

length(unique(genes$gene_id))# so all ensembl ID are unique at this step

genes <- genes[order(genes$gene_id),]


## Subset RNA matrix to keep only protein coding genes
RNA_matrix_TARGET_ALL <- RNA_matrix_TARGET[rownames(RNA_matrix_TARGET) %in% genes$gene_id, ]

tmp<- intersect(genes$gene_id,rownames(RNA_matrix_TARGET_ALL))
tmp <- genes[genes$gene_id %in% tmp,]#subset gene to match the genes in my df

all(tmp$gene_id == rownames(RNA_matrix_TARGET_ALL)) #they are all in the same order of genes
rownames(RNA_matrix_TARGET_ALL) <- tmp$gene_name #rename 

saveRDS(RNA_matrix_TARGET_ALL, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_AcuteLeuk_convert.rds")
```

## ALL obj
```{r}
# Object with only Acute lymphoblasitc thus on P2
clinical_matrix_TARGET_P2  <- readRDS(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinial_matrix_TARGET_P2.rds")
RNA_matrix_TARGET_P2  <- readRDS(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_P2.rds")


# reorgnaisation medatadata
unique(clinical_matrix_TARGET_P2$Classification)
# The classification column is detailed for AML but not for T and B.. I need to sort this into different columns
colnames(clinical_matrix_TARGET_P2) <- c("TARGET_USI","detailed_class")
clinical_matrix_TARGET_P2$simple_class <- clinical_matrix_TARGET_P2$detailed_class
unique(clinical_matrix_TARGET_P2$simple_class)

clinical_matrix_TARGET_P2 <- within(clinical_matrix_TARGET_P2,simple_class[detailed_class %in% c("B-Precursor")] <- "B-ALL")
clinical_matrix_TARGET_P2 <- within(clinical_matrix_TARGET_P2,simple_class[detailed_class %in% c("T Cell ALL")] <- "T-ALL")

#Some category I don"t need and filter out
sum(clinical_matrix_TARGET_P2$simple_class == "B precursor (Non-T, Non-B ALL)", na.rm = TRUE)
clinical_matrix_TARGET_P2 <- clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$simple_class != "B precursor (Non-T, Non-B ALL)",]

# Get subgroup of T-ALL
# Loading the subgroup classification file
P2_Target_clinical_TALL <- readxl::read_xlsx("/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/GDCdata/TARGET-ALL-P2/Clinical/Clinical_Supplement/206f7eec-326e-4fcd-b244-338a522ff384/TARGET_ALL_ClinicalData_Validation_Supplement_20230727.xlsx")
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[,c("TARGET USI","Group")] #subsetting the df

# For T-ALL patient removing from sample info P2 patient with unknow T-ALL subgroup patient
unique(P2_Target_clinical_TALL$Group)
Unk <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "Unknown",]$`TARGET USI`
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[ ! P2_Target_clinical_TALL$`TARGET USI` %in% Unk, ]
Unk <- P2_Target_clinical_TALL[which(is.na(P2_Target_clinical_TALL$Group)),]$`TARGET USI` # Get NA patient
P2_Target_clinical_TALL <- P2_Target_clinical_TALL[ ! P2_Target_clinical_TALL$`TARGET USI` %in% Unk, ]

tal1<- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "TAL1",]$`TARGET USI` # all ID for TAL1 patient
tal2<- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "TAL2",]$`TARGET USI` 
tal3<- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "LMO1/2",]$`TARGET USI` 
NKX1<- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "NKX2_1",]$`TARGET USI`
LYL <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "LMO2_LYL1",]$`TARGET USI` 
HOXA <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "HOXA",]$`TARGET USI` 
TLX1 <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "TLX1",]$`TARGET USI` 
TLX3 <- P2_Target_clinical_TALL[P2_Target_clinical_TALL$Group == "TLX3",]$`TARGET USI`

clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$TARGET_USI %in% tal1,]$detailed_class <- "TAL/LMO"
clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$TARGET_USI %in% tal2,]$detailed_class <- "TAL/LMO"
clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$TARGET_USI %in% tal3,]$detailed_class <- "TAL/LMO"
clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$TARGET_USI %in% NKX1,]$detailed_class <- "TLX"
clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$TARGET_USI %in% LYL,]$detailed_class <- "LYL/LMO"
clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$TARGET_USI %in% HOXA,]$detailed_class <- "HOXA"
clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$TARGET_USI %in% TLX1,]$detailed_class <- "TLX"
clinical_matrix_TARGET_P2[clinical_matrix_TARGET_P2$TARGET_USI %in% TLX3,]$detailed_class <- "TLX"

unique(clinical_matrix_TARGET_P2$detailed_class)

#subset RNA matrix to match clinical ID 
patientID_keep <-intersect(clinical_matrix_TARGET_P2$TARGET_USI,colnames(RNA_matrix_TARGET_P2)) 
RNA_matrix_TARGET_P2<- RNA_matrix_TARGET_P2[,patientID_keep]

saveRDS(clinical_matrix_TARGET_P2, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/clinical_matrix_TARGET_ALL.rds")
saveRDS(RNA_matrix_TARGET_P2, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_ALL.rds")
```

### Convert ID
```{r}
# https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/ to convert geneID using TCGA gtf
#GENCODE v36 gtf https://gdc.cancer.gov/about-data/gdc-data-processing/gdc-reference-files

BiocManager::install("rtracklayer")
library(rtracklayer)
gtf.gr = rtracklayer::import('/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/gencode.v36.annotation.gtf') 
gtf.df = as.data.frame(gtf.gr)

gtf.df <- gtf.df[gtf.df$type == "gene",]
gtf.df <- gtf.df[gtf.df$gene_type == "protein_coding",] #keeping only protein coding Gene

#Looking at duplicates
n_occur <- data.frame(table(gtf.df$gene_name))
temp <- n_occur[n_occur$Freq > 1,]

#Modification to add their Ensmbl ID
tmp <- gtf.df[gtf.df$gene_name %in% temp$Var1,]
tmp <- tmp[,c("gene_id","gene_name")] #subsetting the df
tmp$gene_name <- paste(tmp$gene_id,tmp$gene_name,sep="_")

gtf.df[gtf.df$gene_id %in% tmp$gene_id,]$gene_name <- tmp$gene_name

genes = (gtf.df[ ,c("gene_id","gene_name")])

length(unique(genes$gene_id))# so all ensembl ID are unique at this step

genes <- genes[order(genes$gene_id),]


## Subset RNA matrix to keep only protein coding genes
RNA_matrix_TARGET_P2 <- RNA_matrix_TARGET_P2[rownames(RNA_matrix_TARGET_P2) %in% genes$gene_id, ]

tmp<- intersect(genes$gene_id,rownames(RNA_matrix_TARGET_P2))
tmp <- genes[genes$gene_id %in% tmp,]#subset gene to match the genes in my df if needed

all(tmp$gene_id == rownames(RNA_matrix_TARGET_P2)) #they are all in the same order of genes


rownames(RNA_matrix_TARGET_P2) <- tmp$gene_name #rename 

saveRDS(RNA_matrix_TARGET_P2, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/TCGA/RNA_matrix_TARGET_ALL_convert.rds")
```

