---
output: html_document
editor_options: 
  chunk_output_type: console
---



```{r}
library(Rsubread)
library(DESeq2)
library(ggplot2)
install.packages("ggrepel")
library(ggrepel)
library(clusterProfiler)
library(org.Hs.eg.db)
library(dplyr)
library(fgsea)
library(msigdbr)
library(ComplexHeatmap)
```

Clone recap with genes that are KO: 
D2 et D2bis electroporated with CRISPR guides but no edition

B4 - BEX1
G10 - BEX1
G3 - BEX1
  2B10 - BEX1 et 5
  4C8 - BEX1 et 5
  5B3 - BEX1 et 5
    1D9 - BEX1 et 5 et 2
    2G5 - BEX1 et 5 et 2
    3B6 - BEX1 et 5 et 2
E4n3 - BEX2 et 5
B10n2 - BEX1 et 5
Jurkat - WT
Jurkatbis - WT


Filtrer gene peux exprimer !

#import feature count data
```{r}
count_mtx <- read.table(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/02_Preprocessed/Count/allbam_featurecounts.txt",header = T)
head(count_mtx)

# Prep count matrix for DESeq2
count_mtx <- as.data.frame(count_mtx[,-c(2:6)]) #exclude non informative column
count_mtx2 <- as.data.frame(count_mtx[,-1])
rownames(count_mtx2) <- count_mtx[,1]
names(count_mtx2) <- c("1D9","2B10","2G5","3B6","4C8","5B3","B10n2","B4","D2bis","D2","E4n3","EV","G10","G3","Jktbis","Jkt")
head(count_mtx2)
# remove EV because it's for an other project
count_mtx2 <-count_mtx2[,-12]

#prep sample info matrix
condition <- factor(c("TripleKo","DoubleKo","TripleKo","TripleKo","DoubleKo","DoubleKo","other","SimpleKo","Electroporated","Electroporated","other","SimpleKo","SimpleKo", "WT","WT"))


colData <- data.frame(row.names = colnames(count_mtx2), condition)
colData$condition2 <- condition2 <- factor(c("TripleKo","DoubleKo","TripleKo","TripleKo","DoubleKo","DoubleKo","other","SimpleKo","WT","WT","other","SimpleKo","SimpleKo", "WT","WT")) # to test
```


# Pre-processing obj 1
## Cleaning the data
First obj
```{r}
DESeq.ds <- DESeqDataSetFromMatrix(countData = count_mtx2,
colData = colData,
design = ~ condition)

#Genes with very low counts in all samples provide little evidence for differential expression, it might impact further analysis by adding multiple test in DGE reducing the power of the stat test

# filter inf 10 for 50% samples
nb_sample = 15
keep <- rowSums(counts(DESeq.ds) >= 10) >= nb_sample*0.5
DESeq.ds <- DESeq.ds[keep,]


dds <- DESeq(DESeq.ds)
```


## Jurkat electro
Check on the WT and electroporated Jkt cells, verify if the electroporation induced differncies in gene expression.
```{r}
res <- results(dds, contrast = c("condition", "Electroporated", "WT"))
res_sorted_by_padj <- res[order(res$padj), ]
```

```{r}
df_Ctrl <- as.data.frame(res_sorted_by_padj)
df_Ctrl$absfoldchange <- abs(df_Ctrl$log2FoldChange)
```

## Volcano
```{r}
df_Ctrl$Significant <- ifelse(df_Ctrl$padj < 0.05 & df_Ctrl$absfoldchange >1, "Significative","Not Significative")
dim(na.omit(df_Ctrl[df_Ctrl$Significant == "Significative",]))
df_Ctrl$genename <- rownames(df_Ctrl)

ggplot(df_Ctrl, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(color = Significant))+
   scale_color_manual(values = c("#3A3A3A","#C00000" ,"#3A3A3A")) +
   theme_bw(base_size = 12) + theme(legend.position = "bottom")  +geom_vline(xintercept=c(-1, 1), col="black")+
        geom_hline(yintercept=-log10(0.05), col="black")+
 geom_text_repel(
    data = subset(df_Ctrl, df_Ctrl$padj < 10^-10),
    aes(label = genename),
    size = 4,
    box.padding = unit(0.35, "lines"),
    point.padding = unit(0.3, "lines"))+
  xlim(-26, 26)

min(na.omit(df_Ctrl$log2FoldChange))
max(na.omit(df_Ctrl$log2FoldChange))
```

```{r}
vsd <- vst(dds,blind = FALSE)
plotPCA(vsd, intgroup=c("condition"),pcsToUse = 1:2)


z <- plotPCA(vsd, intgroup=c("condition"),pcsToUse = 1:2)
## replot, obscuring points with the sample name
z + geom_label(aes(label = name))
## or 'nudge' the points up a bit
nudge <- position_nudge(y = 1)
z + geom_label(aes(label = name), position = nudge)
```

Have a look if those very few genes are enriched in a GO terms
## KEGG
```{r}
## KEGG
df_Ctrl$signedpvalue <- ((-sign(df_Ctrl$log2FoldChange) * log10(df_Ctrl$padj)))

organism = "org.Dm.eg.db"
ids<-bitr(rownames(df_Ctrl), fromType = "SYMBOL", toType = "ENTREZID", OrgDb='org.Hs.eg.db')
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

#Create new df with ID
df_KEGG_ctrl = df_Ctrl[df_Ctrl$genename %in% dedup_ids$SYMBOL,]
df_KEGG_ctrl$KEGGID = dedup_ids$ENTREZID

kegg_gene_list <- df_KEGG_ctrl$signedpvalue
names(kegg_gene_list) <- df_KEGG_ctrl$KEGGID
kegg_gene_list<-na.omit(kegg_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)


KEGG_ctrl <-gseKEGG(geneList = kegg_gene_list, organism = "hsa",keyType = "kegg",pvalueCutoff = 0.05)
# no results
```

## Cluster Profiler
```{r}
library(enrichplot)
# foldchange positive means epxression is higer in tripleKo than WT

sigGenes_UP <- na.omit(df_Ctrl[df_Ctrl$padj < 0.05 & df_Ctrl$log2FoldChange > 0 & df_Ctrl$absfoldchange >1,])
sigGenes_DOWN <- na.omit(df_Ctrl[df_Ctrl$padj < 0.05 & df_Ctrl$log2FoldChange < 0 & df_Ctrl$absfoldchange >1 ,])


CPenrich <- enrichGO(gene= rownames(sigGenes_DOWN) , OrgDb = 'org.Hs.eg.db', ont="BP",keyType = "SYMBOL",universe = rownames(df_Ctrl)) 
dotplot(CPenrich,showCategory=20, font.size = 14)
heatplot(CPenrich)


CPenrich <- enrichGO(gene= rownames(sigGenes_UP) , OrgDb = 'org.Hs.eg.db', ont="BP",keyType = "SYMBOL",universe = rownames(df_Ctrl)) 
dotplot(CPenrich,showCategory=20, font.size = 14)
heatplot(CPenrich)

```

## GSEA
```{r}
genelist <- ((-sign(df_Ctrl$log2FoldChange) * log10(df_Ctrl$padj)))
names(genelist) <- df_Ctrl$genename
genelist <- sort(genelist,decreasing = T) #Sort list in decreasing order for Cluster Profiler

#### FGSEA enrichment
### Download specific gene set from Msigdb
msig_C2 <- msigdbr(species = "Homo sapiens", category = "C2") #Currated gene set from paper, pathway...
msig_H <- msigdbr(species = "Homo sapiens", category = "H") # HALLMARK


#Now fixing format to work with fgsea
pathwaysHC2 = split(x = msig_C2$gene_symbol, f = msig_C2$gs_name)
pathwaysH = split(x = msig_H$gene_symbol, f = msig_H$gs_name)


fgseaRes <- fgsea(pathways=pathwaysHC2, stats=genelist,eps=0)
fgseaRes <- fgsea(pathways=pathwaysH, stats=genelist,eps=0)
head(fgseaRes[order(pval), ])
(fgseaRes[order(NES), ])

topPathwaysUp <- fgseaRes[NES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- fgseaRes[NES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
plotGseaTable(pathwaysHC2[topPathways], genelist, fgseaRes, 
              gseaParam = 0.5)

plotGseaTable(pathwaysH[topPathways], genelist, fgseaRes, 
              gseaParam = 0.5)

```

--> No KEGG, GO or GSEA term/pathway enriched
Electroporation don't modify the transcritpome and we can use it as a control

# Pre-processing obj 2
## cleaning the data
Modifying the matrice for DESEQ2
#import feature count data
```{r}
count_mtx <- read.table(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/02_Preprocessed/Count/allbam_featurecounts.txt",header = T)
head(count_mtx)

# Prep count matrix for DESeq2
count_mtx <- as.data.frame(count_mtx[,-c(2:6)]) #exclude non informative column
count_mtx2 <- as.data.frame(count_mtx[,-1])
rownames(count_mtx2) <- count_mtx[,1]
names(count_mtx2) <- c("1D9","2B10","2G5","3B6","4C8","5B3","B10n2","B4","D2bis","D2","E4n3","EV","G10","G3","Jktbis","Jkt")
head(count_mtx2)
# remove EV because it's for an other project
count_mtx2 <-count_mtx2[,-12]

count_mtx2 <-count_mtx2[,-10]
count_mtx2 <-count_mtx2[,-7]
count_mtx2 <-count_mtx2[,-9]

condition <- factor(c("TripleKo","DoubleKo","TripleKo","TripleKo","DoubleKo","DoubleKo","SimpleKo","WT","SimpleKo","SimpleKo", "WT","WT")) #assign D2bis as contrl


colData <- data.frame(row.names = colnames(count_mtx2), condition)
```

```{r}
DESeq.ds <- DESeqDataSetFromMatrix(countData = count_mtx2,
colData = colData,
design = ~ condition)

#Genes with very low counts in all samples provide little evidence for differential expression, it might impact further analysis by adding multiple test in DGE reducing the power of the stat test

# filter inf 10 for 50% samples
nb_sample = 15
keep <- rowSums(counts(DESeq.ds) >= 10) >= nb_sample*0.5
DESeq.ds <- DESeq.ds[keep,]

dds <- DESeq(DESeq.ds)
```

## JKT vs triple
```{r}
res <- results(dds, contrast = c("condition", "WT", "TripleKo"))
res_sorted_by_padj <- res[order(res$padj), ]
```


```{r}
df_DT_KO <- as.data.frame(res_sorted_by_padj)
df_DT_KO$absfoldchange <- abs(df_DT_KO$log2FoldChange)

df_DT_KO$Significant <- ifelse(df_DT_KO$padj < 0.05 & df_DT_KO$absfoldchange >1, "Significative","Not Significative")

dim(na.omit(df_DT_KO[df_DT_KO$Significant == "Significative",]))
```

### Volcano
```{r}
# foldchange negative means epxression si lower in simple KO than WT
# foldchange positive means epxression si higher in simple KO than WT
#### Volcano plot #####

df_DT_KO$genename <- rownames(df_DT_KO)

# define axis size 
plus_grande_valeur <- max(abs(min(na.omit(df_DT_KO$log2FoldChange))), abs(max(na.omit(df_DT_KO$log2FoldChange))))
plus_grande_valeur<- ceiling(plus_grande_valeur) # arrondis a l'entier sup

ggplot(df_DT_KO, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(color = Significant))+
   scale_color_manual(values = c("#3A3A3A","#C00000" ,"#3A3A3A")) +
   theme_bw(base_size = 12) + theme(legend.position = "bottom")  +geom_vline(xintercept=c(-1, 1), col="black")+
        geom_hline(yintercept=-log10(0.05), col="black")+
  xlim(-plus_grande_valeur, plus_grande_valeur)+
 geom_text_repel(
    data = subset(df_DT_KO, df_DT_KO$padj < 10^-4 & df_DT_KO$absfoldchange >1),
    aes(label = genename),
    size = 4,
    box.padding = unit(0.35, "lines"),
    point.padding = unit(0.3, "lines"))
```

### Heatmap
```{r}
vsd <- vst(dds,blind = FALSE)

mat_vst <- assay(vsd)

gene_vec <- na.omit(df_DT_KO[df_DT_KO$Significant == "Significative",])$genename

mat_vst_sub <- mat_vst[gene_vec,]
scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score

hm_colors = c("dodgerblue1","white","red2")
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("B4", "G10", "G3", "2B10","4C8","5B3","1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue", "DoubleKo" = "red","SimpleKo" = "pink","other"="green","Electroporated"="grey","WT"="black"))
)


set.seed(123) # for kmean reproducibility
ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, row_split = 7, column_split = annotation_df$condition)


ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = T,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, row_split = 7)
```



Get genes from ComplexHeatmap clustering
```{r}
ht <- ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = T,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, row_split = 7)

ht_drawn <- draw(ht)
# Get cluster & genes of interest from dendrogram
row_order_list <- row_order(ht_drawn)
gene_clusters <- lapply(row_order_list, function(ind) rownames(scaled_mat)[ind])
gene_clusters[[5]]

write.table(gene_clusters[[5]],file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/cluster5.csv",sep = ";",quote = F,row.names = F,col.names = F)
write.table(gene_clusters[[1]],file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/cluster1.csv",sep = ";",quote = F,row.names = F,col.names = F)
write.table(gene_clusters[[2]],file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/cluster2.csv",sep = ";",quote = F,row.names = F,col.names = F)
write.table(gene_clusters[[3]],file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/cluster3.csv",sep = ";",quote = F,row.names = F,col.names = F)
write.table(gene_clusters[[4]],file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/cluster4.csv",sep = ";",quote = F,row.names = F,col.names = F)
```

# test

```{r}
vsd <- vst(dds,blind = FALSE)
mat_vst <- assay(vsd)

gene_vec <- c("BEX1","BEX2","BEX3","BEX5")


mat_vst_sub <- mat_vst[gene_vec,]

hm_colors = c("dodgerblue1","white","red2")
#to automate scale
hm_limit =c(min(mat_vst_sub),0, max(mat_vst_sub))

column_order <- c("B4", "G10", "G3", "2B10","4C8","5B3","1D9","2G5","3B6","Jktbis","Jkt","D2bis")
mat_vst_sub <- mat_vst_sub[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue", "DoubleKo" = "red","SimpleKo" = "pink","other"="green","Electroporated"="grey","WT"="black"))
)


set.seed(123) # for kmean reproducibility
ComplexHeatmap::Heatmap(mat_vst_sub,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, row_split = 7, column_split = annotation_df$condition,col = rev(rainbow(10)))


breaksList <- seq(-2.5, 2.5, 0.01)
ComplexHeatmap::Heatmap(mat_vst_sub,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, row_split = 7, column_split = annotation_df$condition,col = colorRampPalette(c("#3b0065","#3b0065", "#FFFFFF","#bc0000","#bc0000"))(length(breaksList) - 1))


##### SCALED ####s
scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
hm_colors = c("dodgerblue1","white","red2")
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("B4", "G10", "G3", "2B10","4C8","5B3","1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue", "DoubleKo" = "red","SimpleKo" = "pink","other"="green","Electroporated"="grey","WT"="black"))
)


set.seed(123) # for kmean reproducibility



ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, row_split = 7, column_split = annotation_df$condition)
```

# TAL1

## Get genes
Trying to redo what delphine did online
```{r}
BiocManager::install("ChIPseeker")
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)

# Get bed files and select only JKT https://remap.univ-amu.fr/target_page/TAL1:9606
peak <- readPeakFile("/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/remap2022_TAL1_all_macs2_hg38_v1_0_jurkat.bed")
# Get genes from chip reagion (2 nearest 10Kb)
peakAnno <- annotatePeak(peak = peak, tssRegion = c(-100,100),annoDb="org.Hs.eg.db",TxDb = TxDb.Hsapiens.UCSC.hg38.knownGene)

unique(peakAnno@anno$SYMBOL)

genes_near_peaks <- as.data.frame(peakAnno)$geneId


test <- peakAnno@anno[abs(peakAnno@anno$distanceToTSS) < 5000,]
unique(test$SYMBOL)

#####################
BiocManager::install("rGREAT")
library(rGREAT)

submitGreatJob(gr=peak,species = "hg38",rule = "twoClosest",adv_twoDistance=10000)


great(gr=peak,mode="twoClosest")

test <- genes(TxDb.Hsapiens.UCSC.hg38.knownGene)
overlaps <- findOverlaps(peak,test )
annotated_regions <- test[subjectHits(overlaps)]

gene_symbols <- mapIds(org.Hs.eg.db,
                       keys = annotated_regions$gene_id,           
                       column = "SYMBOL",
                       keytype = "ENTREZID",
                       multiVals = "first") 
unique(gene_symbols)



###################
peak <- readPeakFile("/workspace/NASBioinfo/LALT/BIOINFO/BEX/GSE25000.bed") #PALII
peak2 <- readPeakFile("/workspace/NASBioinfo/LALT/BIOINFO/BEX/GSE29180.bed") # SANDA

peakAnno <- annotatePeak(peak = peak, tssRegion = c(-3000,3000),annoDb="org.Hs.eg.db",TxDb = TxDb.Hsapiens.UCSC.hg38.knownGene)

peakAnno2 <- annotatePeak(peak = peak2, tssRegion = c(-3000,3000),annoDb="org.Hs.eg.db",TxDb = TxDb.Hsapiens.UCSC.hg38.knownGene)

t1<- intersect(peakAnno@anno$SYMBOL,gene_julie)
t2 <-intersect(peakAnno2@anno$SYMBOL,gene_julie)

intersect(t1,t2)


length(unique(peakAnno2@anno$SYMBOL))
length(unique(peakAnno@anno$SYMBOL))

## filtering 10kb
test <- peakAnno@anno[abs(peakAnno@anno$distanceToTSS) < 10000,]
length(unique(test$SYMBOL))

test2_10kb <- peakAnno2@anno[abs(peakAnno2@anno$distanceToTSS) < 10000,]
length(unique(test2_10kb$SYMBOL))


t1<- intersect(test$SYMBOL,gene_julie)
t2 <-intersect(test2_10kb$SYMBOL,gene_julie)

intersect(t1,t2)
# We choose SANDA based on genes that stay in the intersection

#some gene are multiple time to catch all their name
gene_julie <-read.table(file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/19_Users/Julie/gene_cible_TAL1.txt")
manual_domie <- c("RAG1","ETV6","CDK6","GATA1","GATA2","TRIB2","TAL1","CD69","RUNX1","NFKB1D","CD5")# add from DOMIE
gene_julie <- unique(c(gene_julie$V1,manual_domie))


## Control negative
#MED1 dans JKT
peak_MED1 <- readPeakFile("/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/GSE59657.MED1.Jurkat.bed")
peakAnno_MED1 <- annotatePeak(peak = peak_MED1, tssRegion = c(-3000,3000),annoDb="org.Hs.eg.db",TxDb = TxDb.Hsapiens.UCSC.hg38.knownGene)
length(unique(peakAnno_MED1@anno$SYMBOL))

## filtering 10kb
MED1_10kb <- peakAnno_MED1@anno[abs(peakAnno_MED1@anno$distanceToTSS) < 10000,]
length(unique(MED1_10kb$SYMBOL))
```

Heatmaps test of those genes
```{r}
gene_remap_10kb <- unique(test2_10kb$SYMBOL) #SANDA 10KB filtering
gene_remap <- unique(peakAnno2@anno$SYMBOL) # SANDA

MED1_10kb <- MED1_10kb$SYMBOL

## New heatmap with only highly variabel genes
vsd <- vst(dds,blind = FALSE)
mat_vst <- assay(vsd)

#topVarGenes <- head(order(rowVars(mat_vst), decreasing = TRUE), 2000)
#tmp_var <- mat_vst[topVarGenes,]

gene_vec <- gene_remap_10kb
gene_vec <- gene_remap
gene_vec <- MED1_10kb
gene_vec <- rownames(dds)
vec_reduit <- gene_vec[gene_vec %in% rownames(mat_vst)]

#gene_interest <- intersect(rownames(tmp_var),vec_reduit)

mat_vst_sub <- mat_vst[vec_reduit,]

scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)

set.seed(123) # for kmean reproducibility

ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_split = 4,row_names_gp = gpar(fontsize = 8))

```
Heatmaps test of those genes with interseet variable gene

```{r}
gene_remap_10kb <- unique(test2_10kb$SYMBOL) #SANDA 10KB filtering
gene_remap <- unique(peakAnno2@anno$SYMBOL) # SANDA

## New heatmap with only highly variabel genes
vsd <- vst(dds,blind = FALSE)
mat_vst <- assay(vsd)

topVarGenes <- head(order(rowVars(mat_vst), decreasing = TRUE), 2000)
tmp_var <- mat_vst[topVarGenes,]

gene_vec <- gene_remap_10kb
#gene_vec <- gene_remap
vec_reduit <- gene_vec[gene_vec %in% rownames(mat_vst)]

gene_interest <- intersect(rownames(tmp_var),vec_reduit)

mat_vst_sub <- mat_vst[gene_interest,]

scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)

set.seed(123) # for kmean reproducibility

ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_names_gp = gpar(fontsize = 8))
```

Heatmaps test of those genes with interseet variable gene from DGE
# plot papier ?
```{r}
res <- results(dds, contrast = c("condition", "WT", "TripleKo"))

df_test <- as.data.frame(res)
df_test$absfoldchange <- abs(df_test$log2FoldChange)

df_test$Significant <- ifelse(df_test$padj < 0.05 & df_test$absfoldchange >1, "Significative","Not Significative")

significant_gene <- rownames(na.omit(df_test[df_test$Significant == "Significative",]))
#significant_table <- (na.omit(df_test[df_test$Significant == "Significative",]))

gene_remap_10kb <- read.table(file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/sanda_updatejulie.csv",sep=";") # manually modif by domie and julie
gene_remap_10kb <- gene_remap_10kb $V1

## New heatmap with only highly variabel genes
vsd <- vst(dds,blind = FALSE)
mat_vst <- assay(vsd)

gene_vec <- gene_remap_10kb
#gene_vec <- gene_remap
vec_reduit <- gene_vec[gene_vec %in% rownames(mat_vst)]

gene_interest2 <- intersect(significant_gene,vec_reduit)


#write.table(vec_reduit,file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/gene_SANDA.txt",quote = F)


mat_vst_sub <- mat_vst[gene_interest2,]

scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)

set.seed(123) # for kmean reproducibility

ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_names_gp = gpar(fontsize = 6),row_split = 2)

png(filename = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/DGE_SANDA_Figure.png", width = 1500, height = 2500,res= 200)
plot(ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_names_gp = gpar(fontsize = 6),row_split = 2))
dev.off()

pdf("/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/DGE_SANDA_Figure.pdf" ,width = 7, height = 13)
plot(ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_names_gp = gpar(fontsize = 6),row_split = 2))
dev.off()

#gene_interest2[gene_interest2 == "CD5"]

# list domie
tmp <- df_test[rownames(df_test) %in% gene_interest2,]
tmp2 <- tmp[tmp$log2FoldChange > 1,] #UP WT
tmp3 <- tmp[tmp$log2FoldChange < 1,] #UP Triple KO

write.table(tmp2,file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/gene_TAL1_UP_WT_V2.txt",quote = F)
write.table(tmp3,file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/gene_TAL1_UP_TRIPLEKO_V2.txt",quote = F)

```

# plot 25

```{r}

res <- results(dds, contrast = c("condition", "WT", "TripleKo"))

df_test <- as.data.frame(res)
df_test$absfoldchange <- abs(df_test$log2FoldChange)

df_test$Significant <- ifelse(df_test$padj < 0.05 & df_test$absfoldchange >1, "Significative","Not Significative")

significant_gene <- rownames(na.omit(df_test[df_test$Significant == "Significative",]))
#significant_table <- (na.omit(df_test[df_test$Significant == "Significative",]))

## New heatmap with only highly variabel genes
vsd <- vst(dds,blind = FALSE)
mat_vst <- assay(vsd)

gene_vec <- c("CEBPE","SPINK2","CHRNA9","STAT5A","SIX6","SVOPL","CCR4","ANXA3","ERG","ADCYAP1","ABCB10","CD69","GIMAP6","GIMAP2","PTCRA","NOTCH3","KCNJ4","GRID1","CD2","CD5","GATA2","GIMAP4","KIT","TRIB2","CDKN1A") # list domie
#gene_vec <- gene_remap
vec_reduit <- gene_vec[gene_vec %in% rownames(mat_vst)]

gene_interest2 <- intersect(significant_gene,vec_reduit)


#write.table(vec_reduit,file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/gene_SANDA.txt",quote = F)


mat_vst_sub <- mat_vst[gene_interest2,]

scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)

set.seed(123) # for kmean reproducibility

ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_names_gp = gpar(fontsize = 11),row_split = 2)

```



test enrichissement est ce que dans les genes DGE on est enrihcit ou pas en gene de la liste 10KB sanda ? 
```{r}
library(stats)
  
group2 <- significant_gene
sanda_group <- intersect(gene_remap_10kb,rownames(dds))
overlapp <- intersect(sanda_group,significant_gene)
total_gene<- rownames(dds)



phyper(length(overlapp)-1,length(group2),length(total_gene)- length(group2),length(sanda_group),lower.tail = F)
# pvalue ok mais pas dingue
```

test enrichissement fonctionnel
```{r}
## GET GENES ###
ht <- ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_names_gp = gpar(fontsize = 8),row_split = 4)
ht_drawn <- draw(ht)
# Get cluster & genes of interest from dendrogram
row_order_list <- row_order(ht_drawn)
gene_clusters <- lapply(row_order_list, function(ind) rownames(scaled_mat)[ind])
gene_clusters[[2]]



## GENE ONTOLOGY ##
CPenrich <- enrichGO(gene= gene_clusters[[2]] , OrgDb = 'org.Hs.eg.db', ont="BP",keyType = "SYMBOL",universe = rownames(dds)) 
dotplot(CPenrich,showCategory=20, font.size = 20)

CPenrich <- enrichGO(gene= gene_clusters[[4]] , OrgDb = 'org.Hs.eg.db', ont="BP",keyType = "SYMBOL",universe = rownames(dds)) 
dotplot(CPenrich,showCategory=20, font.size = 14)



## From ALL
ht <- ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_names_gp = gpar(fontsize = 8),row_split = 2)
ht_drawn <- draw(ht)
# Get cluster & genes of interest from dendrogram
row_order_list <- row_order(ht_drawn)
gene_clusters <- lapply(row_order_list, function(ind) rownames(scaled_mat)[ind])
gene_clusters[[2]]

## GENE ONTOLOGY ##
CPenrich <- enrichGO(gene= gene_clusters[[1]] , OrgDb = 'org.Hs.eg.db', ont="BP",keyType = "SYMBOL",universe = rownames(dds)) 
dotplot(CPenrich,showCategory=20, font.size = 20)

CPenrich <- enrichGO(gene= gene_clusters[[2]] , OrgDb = 'org.Hs.eg.db', ont="BP",keyType = "SYMBOL",universe = rownames(dds)) 
dotplot(CPenrich,showCategory=20, font.size = 20)
heatplot(CPenrich, showCategory=10)

# I will use the ALL list and not 10kb
write.table(gene_clusters[[1]],file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/LIST_ALL_C1.txt",quote = F,row.names = F,col.names = F)
write.table(gene_clusters[[2]],file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/LIST_ALL_C2.txt",quote = F,row.names = F,col.names = F)


## Icis target on DGE without filtering by TAL chip
DGE_data <- (na.omit(df_test[df_test$Significant == "Significative",]))
write.table(rownames(DGE_data),file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/LIST_DGE_complete.txt",quote = F,row.names = F,col.names = F)

DGE_data[DGE_data$log2FoldChange > 1,]
write.table(rownames(DGE_data[DGE_data$log2FoldChange > 1,]),file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/LIST_DGE_UP.txt",quote = F,row.names = F,col.names = F)

DGE_data[DGE_data$log2FoldChange < 1,]
write.table(rownames(DGE_data[DGE_data$log2FoldChange < 1,]),file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/LIST_DGE_DOWN.txt",quote = F,row.names = F,col.names = F)

```

test icis target with peak region of TAL1
```{r}
bed_UP_WT<- peakAnno2@anno[peakAnno2@anno$SYMBOL %in%gene_clusters[[1]] ,]
mcols(bed_UP_WT) <- NULL

bed_UP_WT <- as.data.frame(bed_UP_WT)
bed_UP_WT <- bed_UP_WT[1:3]
write.table(bed_UP_WT, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/CHIP_UP_WT.bed",quote = F,sep = "\t",row.names = F,col.names = F)



bed_UP_TRIPLEKO<- peakAnno2@anno[peakAnno2@anno$SYMBOL %in%gene_clusters[[2]] ,]
mcols(bed_UP_TRIPLEKO) <- NULL
bed_UP_TRIPLEKO <- as.data.frame(bed_UP_TRIPLEKO)
bed_UP_TRIPLEKO <- bed_UP_TRIPLEKO[1:3]

write.table(bed_UP_TRIPLEKO, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/CHIP_UP_TRIPLEKO.bed",quote = F,sep = "\t",row.names = F,col.names = F)


## bed all peak
BEDpeak2 <- readPeakFile("/workspace/NASBioinfo/LALT/BIOINFO/BEX/GSE29180.bed") # SANDA
mcols(BEDpeak2) <- NULL
BEDpeak2 <- as.data.frame(BEDpeak2)
BEDpeak2 <- BEDpeak2[1:3]
write.table(BEDpeak2, file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/CHIP_ALL_PEAK.bed",quote = F,sep = "\t",row.names = F,col.names = F)

```


## new matrice
Avec old matrice on prend toute la matrice en compte donc on va avoir des gènes var issu de DoubleKo… alors qu’on s’en désintéresse docn ici refait un obj DESEQ2 avec seulement les 6 samples d'interet


```{r}
count_mtx <- read.table(file = "/workspace/NASBioinfo/LALT/BIOINFO/BEX/02_Preprocessed/Count/allbam_featurecounts.txt",header = T)
head(count_mtx)

# Prep count matrix for DESeq2
count_mtx <- as.data.frame(count_mtx[,-c(2:6)]) #exclude non informative column
count_mtx2 <- as.data.frame(count_mtx[,-1])
rownames(count_mtx2) <- count_mtx[,1]
names(count_mtx2) <- c("1D9","2B10","2G5","3B6","4C8","5B3","B10n2","B4","D2bis","D2","E4n3","EV","G10","G3","Jktbis","Jkt")
head(count_mtx2)
# remove EV because it's for an other project
count_mtx2 <-count_mtx2[,-12]

count_mtx2 <-count_mtx2[,-10]
count_mtx2 <-count_mtx2[,-7]
count_mtx2 <-count_mtx2[,-9]
count_mtx2 <-count_mtx2[,-2]
count_mtx2 <-count_mtx2[,-4]
count_mtx2 <-count_mtx2[,-4]
count_mtx2 <-count_mtx2[,-4]
count_mtx2 <-count_mtx2[,-5]
count_mtx2 <-count_mtx2[,-5]
## il faudrait faire ca en une fois a terme
head(count_mtx2)

condition <- factor(c("TripleKo","TripleKo","TripleKo","WT", "WT","WT")) #assign D2bis as contrl


colData <- data.frame(row.names = colnames(count_mtx2), condition)

DESeq.ds <- DESeqDataSetFromMatrix(countData = count_mtx2,
colData = colData,
design = ~ condition)

#Genes with very low counts in all samples provide little evidence for differential expression, it might impact further analysis by adding multiple test in DGE reducing the power of the stat test

# filter inf 10 for 50% samples
nb_sample = length(condition)
keep <- rowSums(counts(DESeq.ds) >= 10) >= nb_sample*0.5
DESeq.ds <- DESeq.ds[keep,]


dds <- DESeq(DESeq.ds)
```

```{r}
gene_remap <-read.table("/workspace/NASBioinfo/LALT/BIOINFO/BEX/remap2022_TAL1_all_macs2_hg38_v1_0_jurkat_GREAT_2nearest_10kb_associated_genes")

## New heatmap with only highly variabel genes
vsd <- vst(dds,blind = FALSE)
mat_vst <- assay(vsd)

topVarGenes <- head(order(rowVars(mat_vst), decreasing = TRUE), 2000)
tmp_var <- mat_vst[topVarGenes,]


gene_vec <- gene_remap$V1
vec_reduit <- gene_vec[gene_vec %in% rownames(mat_vst)]

gene_interest <- intersect(rownames(tmp_var),vec_reduit)

mat_vst_sub <- mat_vst[gene_interest,]

scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)

set.seed(123) # for kmean reproducibility

ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_split = 2,row_names_gp = gpar(fontsize = 8))



### same type of plot but using DGE instead of top gene
res <- results(dds, contrast = c("condition", "WT", "TripleKo"))

df_test <- as.data.frame(res)
df_test$absfoldchange <- abs(df_test$log2FoldChange)

df_test$Significant <- ifelse(df_test$padj < 0.05 & df_test$absfoldchange >1, "Significative","Not Significative")

significant_gene <- rownames(na.omit(df_test[df_test$Significant == "Significative",]))
significant_table <- (na.omit(df_test[df_test$Significant == "Significative",]))

write.table(significant_table, file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/DGE_WT_TripleKO.txt",quote = F,sep = "\t")



gene_interest2 <- intersect(significant_gene,vec_reduit)
mat_vst_sub <- mat_vst[gene_interest2,]

scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)

set.seed(123) # for kmean reproducibility

ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_split = 2,row_names_gp = gpar(fontsize = 8))

####
intersect(gene_interest,gene_interest2)
```



Get genes from ComplexHeatmap clustering
```{r}
ht <- ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_split = 2,row_names_gp = gpar(fontsize = 8))
ht_drawn <- draw(ht)
# Get cluster & genes of interest from dendrogram
row_order_list <- row_order(ht_drawn)
gene_clusters <- lapply(row_order_list, function(ind) rownames(scaled_mat)[ind])
gene_clusters[[1]]

write.table(gene_clusters[[1]], file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/gene_TAL1_UP_WT.txt",quote = F,row.names = F,col.names = F)
write.table(gene_clusters[[2]], file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/gene_TAL1_UP_TripleKO.txt",quote = F,row.names = F,col.names = F)
```

```{r}
toString(gene_clusters[[2]])
toString(rownames(mat_vst))
```


```{r}
# Ne prend pas en compte la signed p value.. car ici petite liste qui va tous dans le meme sens

organism = "org.Hm.eg.db"
ids<-bitr(gene_clusters[[2]], fromType = "SYMBOL", toType = "ENTREZID", OrgDb='org.Hs.eg.db')
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]


kegg_up <- enrichKEGG(gene = dedup_ids$ENTREZID,
                      organism = 'hsa',    # 'hsa' pour Homo sapiens, adapter selon ton espèce
                      keyType = "kegg",    # Spécifie le type d'identifiants (ici ENTREZ)
                      pvalueCutoff = 0.05) # Ajuste le seuil de p-value si nécessaire

# Afficher les résultats de l'enrichissement
head(kegg_up)



organism = "org.Hm.eg.db"
ids<-bitr(gene_clusters[[1]], fromType = "SYMBOL", toType = "ENTREZID", OrgDb='org.Hs.eg.db')
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]


kegg_up <- enrichKEGG(gene = dedup_ids$ENTREZID,
                      organism = 'hsa',    # 'hsa' pour Homo sapiens, adapter selon ton espèce
                      keyType = "kegg",    # Spécifie le type d'identifiants (ici ENTREZ)
                      pvalueCutoff = 0.05) # Ajuste le seuil de p-value si nécessaire

# Afficher les résultats de l'enrichissement
head(kegg_up)

```


```{r}
CPenrich <- enrichGO(gene= gene_clusters[[2]] , OrgDb = 'org.Hs.eg.db', ont="BP",keyType = "SYMBOL",universe = rownames(df_test)) 
dotplot(CPenrich,showCategory=20, font.size = 14)
```



# GENE JULIE
```{r}
#some gene are multiple time to catch all their name
gene_julie <-read.table(file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/19_Users/Julie/gene_cible_TAL1.txt")
manual_domie <- c("RAG1","ETV6","CDK6","GATA1","GATA2","TRIB2","TAL1","CD69","RUNX1","NFKB1D","CD5")# add from DOMIE

gene_julie <- unique(c(gene_julie$V1,manual_domie))

## test with Variable genes

vsd <- vst(dds,blind = FALSE)
mat_vst <- assay(vsd)

topVarGenes <- head(order(rowVars(mat_vst), decreasing = TRUE), 2000)
tmp_var <- mat_vst[topVarGenes,]


vec_reduit <- gene_julie[gene_julie %in% rownames(mat_vst)]

gene_interest <- intersect(rownames(tmp_var),vec_reduit)

mat_vst_sub <- mat_vst[gene_interest,]

scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)

set.seed(123) # for kmean reproducibility

ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_split = 2,row_names_gp = gpar(fontsize = 10))

## test with DGE
res <- results(dds, contrast = c("condition", "WT", "TripleKo"))

df_test <- as.data.frame(res)
df_test$absfoldchange <- abs(df_test$log2FoldChange)

df_test$Significant <- ifelse(df_test$padj < 0.05 & df_test$absfoldchange >1, "Significative","Not Significative")

significant_gene <- rownames(na.omit(df_test[df_test$Significant == "Significative",]))

gene_interest2 <- intersect(significant_gene,vec_reduit)

mat_vst_sub <- mat_vst[gene_interest2,]

scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)

set.seed(123) # for kmean reproducibility

ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition,row_split = 2,row_names_gp = gpar(fontsize = 10))
```

# SANDA
```{r}
# Load GROUP FROM SANDA Paper https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11063860/
SANDA_A <- read.table(file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/SANDA_GR_A.txt",header = T)
SANDA_B <- read.table(file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/SANDA_GR_B.txt",header = T)
SANDA_C <- read.table(file = "/workspace/userDUPREZ/Equipes_PF_Services/EQ_Duprez/EQ_Duprez/LAL-T/7_Papier/Papier BEX/SANDA_GR_C.txt",header = T)
```

## Heatmaps
```{r}
vsd <- vst(dds,blind = FALSE)

mat_vst <- assay(vsd)

gene_vec <- SANDA_A$GROUPA
gene_interest <- intersect(rownames(mat_vst),gene_vec)


mat_vst_sub <- mat_vst[gene_interest,]
#scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
scaled_mat <- mat_vst_sub

#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)


set.seed(123) # for kmean reproducibility
heatmap1 <- ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition) #col = rev(rainbow(10))

#####
mat_vst <- assay(vsd)

gene_vec <- SANDA_B$GroupB
gene_interest <- intersect(rownames(mat_vst),gene_vec)


mat_vst_sub <- mat_vst[gene_interest,]
#scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
scaled_mat <- mat_vst_sub

#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)


set.seed(123) # for kmean reproducibility
heatmap2<- ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition)

###
mat_vst <- assay(vsd)

gene_vec <- SANDA_C$GroupC
gene_interest <- intersect(rownames(mat_vst),gene_vec)


mat_vst_sub <- mat_vst[gene_interest,]
#scaled_mat = t(scale(t(mat_vst_sub))) # row centered matrix Z score
scaled_mat <- mat_vst_sub

#to automate scale
hm_limit =c(min(scaled_mat),0, max(scaled_mat))

column_order <- c("1D9","2G5","3B6","Jktbis","Jkt","D2bis")
scaled_mat <- scaled_mat[, column_order]

annotation_df <- colData
annotation_df <- annotation_df[column_order, , drop = FALSE]
column_annotation <- HeatmapAnnotation(
  df = annotation_df,
  col = list(condition = c("TripleKo" = "blue","WT"="black"))
)


set.seed(123) # for kmean reproducibility
heatmap3<- ComplexHeatmap::Heatmap(scaled_mat,show_column_names = T ,cluster_columns = F,cluster_rows = T,column_names_side = "top",top_annotation = column_annotation, column_split = annotation_df$condition)


library(gridExtra)
grid.newpage()

# Disposer les heatmaps côte à côte
pushViewport(viewport(layout = grid.layout(1, 3))) # 1 ligne, 2 colonnes
pushViewport(viewport(layout.pos.row = 1, layout.pos.col = 1))
draw(heatmap1,newpage = FALSE)
upViewport()
pushViewport(viewport(layout.pos.row = 1, layout.pos.col = 2))
draw(heatmap2,newpage = FALSE)
upViewport()
pushViewport(viewport(layout.pos.row = 1, layout.pos.col = 3))
draw(heatmap3,newpage = FALSE)
```


## Boxplot
```{r}
res <- results(dds, contrast = c("condition", "WT", "TripleKo"))
test <- as.data.frame(res)

test$signedpvalue <- ((-sign(test$log2FoldChange) * log10(test$pvalue)))
test$log10pval <-  log10(test$pvalue)
test$genes <- rownames(test)


test1 <- test[test$genes %in% SANDA_A$GROUPA,]
boxplot(test1$log10pval,ylim =c(0,-25),main="GROUP A")

test2 <- test[test$genes %in% SANDA_B$GroupB,]
boxplot(test2$log10pval,ylim =c(0,-25),main="GROUP B")

test3 <- test[test$genes %in% SANDA_C$GroupC,]
boxplot(test3$log10pval, ylim =c(0,-25),main="GROUP C")



boxplot(abs(test1$log2FoldChange),ylim =c(0,5),main="GROUP A")
boxplot(abs(test2$log2FoldChange),ylim =c(0,5),main="GROUP B")
boxplot(abs(test3$log2FoldChange), ylim =c(0,5),main="GROUP C")

```



A faire volcano + groupe A highlight ou alors heatmap type https://jokergoo.github.io/ComplexHeatmap-reference/book/a-list-of-heatmaps.html ou on ajoute une legende en plus de côté
## Volcano
On va identifier les genes des groupes sur le volcano en les colorants
```{r}
# define axis size 
plus_grande_valeur <- max(abs(min(na.omit(test$log2FoldChange))), abs(max(na.omit(test$log2FoldChange))))
plus_grande_valeur<- ceiling(plus_grande_valeur) # arrondis a l'entier sup

# subset GROUP SANDA
test1 <- test[test$genes %in% SANDA_A$GROUPA,]
test2 <- test[test$genes %in% SANDA_B$GroupB,]
test3 <- test[test$genes %in% SANDA_C$GroupC,]

ggplot(test, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(color = Significant),size=0.5)+
   scale_color_manual(values = c("azure3","#3A3A3A" ,"azure3")) +
   theme_bw(base_size = 12) + theme(legend.position = "bottom")  +geom_vline(xintercept=c(-1, 1), col="black")+
        geom_hline(yintercept=-log10(0.05), col="black")+
  xlim(-plus_grande_valeur, plus_grande_valeur) + geom_point(data = test2, aes(x = log2FoldChange, y = -log10(padj)), color = "darkturquoise",size=2,alpha = 0.7)+ geom_point(data = test3, aes(x = log2FoldChange, y = -log10(padj)), color = "goldenrod1",size=2,alpha = 0.7)+ geom_point(data = test1, aes(x = log2FoldChange, y = -log10(padj)), color = "brown2",size=2,alpha = 0.7)
```

Enrichissement
Gene SANDA A B ou C est ce qu'il sont sur represente dans le DGE wt triple KO ??
```{r}
library(stats)

test
test$absfoldchange <- abs(test$log2FoldChange)
test$Significant <- ifelse(test$padj < 0.05 & test$absfoldchange >1, "Significative","Not Significative")

gene_vec <- na.omit(test[test$Significant == "Significative",])$genes

group2 <- gene_vec
sanda_group <- intersect(SANDA_A$GROUPA,rownames(dds))

overlapp <- intersect(sanda_group,significant_gene)
total_gene<- rownames(dds)



phyper(length(overlapp)-1,length(group2),length(total_gene)- length(group2),length(sanda_group),lower.tail = F)
# pvalue ok mais pas dingue
(length(overlapp)/length(sanda_group))/(length(group2)/length(total_gene))



group2 <- gene_vec
sanda_group <- intersect(SANDA_B$GroupB,rownames(dds))

overlapp <- intersect(sanda_group,significant_gene)
total_gene<- rownames(dds)

phyper(length(overlapp)-1,length(group2),length(total_gene)- length(group2),length(sanda_group),lower.tail = F)
(length(overlapp)/length(sanda_group))/(length(group2)/length(total_gene))


#####
group2 <- gene_vec
sanda_group <- intersect(SANDA_C$GroupC,rownames(dds))

overlapp <- intersect(sanda_group,significant_gene)
total_gene<- rownames(dds)

phyper(length(overlapp)-1,length(group2),length(total_gene)- length(group2),length(sanda_group),lower.tail = F)
(length(overlapp)/length(sanda_group))/(length(group2)/length(total_gene))


# enrichment calcul from gorilla
#(intersection / SANDA) / (DGE/N)
```

## GSEA
Test enrich de A, B et C
```{r}
genelist <- ((-sign(test$log2FoldChange) * log10(test$padj)))
names(genelist) <- test$genes
genelist <- sort(genelist,decreasing = T) #Sort list in decreasing order for Cluster Profiler

#### FGSEA enrichment


fgseaRes <- fgsea(pathways=pathwaysHC2, stats=genelist,eps=0)
fgseaRes <- fgsea(pathways=pathwaysH, stats=genelist,eps=0)



head(fgseaRes[order(pval), ])
(fgseaRes[order(NES), ])

topPathwaysUp <- fgseaRes[NES > 0][head(order(pval), n=10), pathway]
topPathwaysDown <- fgseaRes[NES < 0][head(order(pval), n=10), pathway]
topPathways <- c(topPathwaysUp, rev(topPathwaysDown))
plotGseaTable(pathwaysHC2[topPathways], genelist, fgseaRes, 
              gseaParam = 0.5)

plotGseaTable(pathwaysH[topPathways], genelist, fgseaRes, 
              gseaParam = 0.5)
```

